### ISSUE TITLE
[FEATURE REQUEST] Text Editor

### ISSUE THREAD

**Issue by nevumx** [d0k3/GodMode9#861]
Similar to how the hex editor can change individual bytes, it would be nice if the Text Viewer had an "edit" mode where new characters could be inserted with the existing software keyboard that is used to name/rename files. (Perhaps an "enter" key might need to be added) I assume that the limitation here is that while it is easy to change bytes, adding them to a larger text file might be problematic from a memory allocation standpoint, so perhaps a limit of 10 MB or so could be imposed, depending on the memory model used by GM9.

**Comment by Wolfvak** [comment]
Already discussed and rejected, please see #117 and #805

**Comment by nevumx** [comment]
Okay, I see those threads now, that being said, I decided to just go forward and implement it myself. The text viewer code is indeed a PITA, but I have already implemented UTF-8 support and a cursor with U/D/L/R movement. I am going to work on moving the window with the cursor, and integrating the keyboard with text insertion/deletion next, but ideally I should be able to make this happen without mutilating the text viewer code too much. I'll keep you all posted...
